### PR TITLE
Add GravityToggleStyle

### PR DESCRIPTION
**Pull request by djalfirevic**

### CODE REVIEW

File: Sources/Props/Toggle/GravityToggleStyle.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    // MARK: - Initializer
+    public init(onLabel: String = "ON", offLabel: String = "OFF") {
+        self.onLabel = onLabel
+        self.offLabel = offLabel
+    }
+    
+    // MARK: - ToggleStyle
+    public func makeBody(configuration: Self.Configuration) -> some View {
+        GeometryReader { geometry in
+            let offset = geometry.size.width/2 - 3 * geometry.size.height/2 + 10
```
Comment (djalfirevic) [inline Sources/Props/Toggle/GravityToggleStyle.swift]: Maybe you have some better math for this. I don't like this magic 10, but it seems that it's working 🙏

Comment (qmchenry) [reply Sources/Props/Toggle/GravityToggleStyle.swift]: It's looking great! I recommend making the size fixed, roughly the size of the built-in toggle, but maybe a little longer to give the effect a little more oomph. If you fix the size, you shouldn't need geometry reading which will simplify, you can decide on the size, then know how big the magnitude of the animation. 

Also as a UX thing, it'd be great if the whole thing was tappable, not just the circle.

**Comment by qmchenry** [comment]
Check out my changes and see if you are into them. Let's add enum types for all new styles while we're implementing them plus some documentation following this example.

File: Catalog/Shared/Toggles/ToggleStyles.swift [hunk 2]
```diff
@@ -46,12 +47,28 @@ struct SunMoonToggleSample: PropSampleable {
     }
 }
 
+struct GravityToggleSample: PropSampleable {
+    let name = "gravity toggle style"
+    let notes: String? = ".toggleStyle(.gravity)"
+    let source = "Custom"
+    let keywords = "toggle custom props animated gravity"
```
Comment (qmchenry) [inline Catalog/Shared/Toggles/ToggleStyles.swift:54]: 'props" here was "third party" which referenced the sun/moon toggle above which is literally a 3rd party SPM imported into the catalog. Let's keep it consistent when we're using our props. Keep an eye on mine, too, as I'm likely to do this, too. :D

Comment (djalfirevic) [reply Catalog/Shared/Toggles/ToggleStyles.swift:54]: Oh right! I see. We should also change the source to be "Props" and not "Custom" then, to be consistent with others?

I've used your changes and applied to that other PR, but we'll check it out after this one ;)

**Review by qmchenry — APPROVED** [review]

**Pull request merged**